### ISSUE TITLE
absRefPrefix on shortcut page

### ISSUE THREAD

**Issue by Guigue79** [dmitryd/typo3-realurl#548]
Hi.
I have this error when users try to access to a shortcut page.
Thu, 12 Oct 2017 09:10:27 +0200 [WARNING] request="54759cfb33cc4" component="DmitryDulepov.Realurl.Encoder.UrlEncoder": config.absRefPrefix is not set! Please, check your TypoScript configuration! 
The config.absRefPrefix is set in the root page.

**Comment by Patsone** [comment]
Same here. My TS contains config.absRefPrefix = / but I receive the same warnings filling up my logfile.

**Comment by hannesbochmann** [comment]
The same here.

**Comment by h3nn3s** [comment]
Jeb, same here. Looks like it is only fired if rootpage is a shortcut

**Comment by bigahuna** [comment]
Hi, we can confirm that we also have this error if rootpage is a shortcut to another page.

**Comment by DiablosOffens** [comment]
I've got the same problem and in combination with some SQL injection attacks (where the "L" parameter cannot be parsed by validateLanguageParameter in UrlEncoder.php) the logfile used up the whole disk space!

**Comment by AndreasA** [comment]
I can also confirm this warning.
The issue is that checkPageForShortcutRedirect is executed prior to setAbsRefPrefix 
which means that the URL encoder is executed before TSFE->absRefPrefix is set which is OK here because TYPO3 takes care to append the current host information afterwards.

**Comment by adrienjacob** [comment]
Hmmmm.... I also have this issue (but not the one with L parameter, maybe an issue with addQueryString or config.linkVars).
Anyway, this absRefPrefix warning seems impossible to avoid when using shortcuts : we cannot access protected property `TypoScriptFrontendController::$originalShortcutPage` to know if we are currently following a shortcut. So maybe we should just remove this warning logging ?

**Comment by fazzyx** [comment]
@markuspoerschke There is a devlog switch, maybe depend the logging on it.

**Comment by AndreasA** [comment]
@markuspoerschke I checked the code and in 7.6, 8.x and current master immediately after absRefPrefix is set the cObj in TSFE is created by executing newCObj. So you could also check regarding that.

**Comment by ervaude** [comment]
You can deactivate the logging of warnings from this class with

```
$GLOBALS['TYPO3_CONF_VARS']['LOG']['DmitryDulepov']['Realurl']['Encoder']['writerConfiguration'] = [
        \TYPO3\CMS\Core\Log\LogLevel::WARNING => []
];
```

**Comment by josefglatz** [comment]
I can also confirm the issue. I've set `/` as value.

**Comment by chrisprog20** [comment]
This happens if 
`config.absRefPrefix = / `
is set.

I tested this in TYPO 7.6.23 and this happened after upgrading to realurl 2.3.1.

I think it has to do with commit [54c5233](https://github.com/dmitryd/typo3-realurl/commit/54c5233041f7ba272ac61a6c2c2f08950294c0c8) (fixing #548 & #570).

If I set

`config.absRefPrefix >`

the error does not happen but then I get again an entry in typo3temp/logs 
giving the message like in #548 or #570.
